File: WS_JAVA_REST_BanQuito/src/main/java/ec/edu/monster/model/CuotaAmortizacion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.model;

/**
 *
 * @author danie
 */

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "cuota_amortizacion")
public class CuotaAmortizacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "id_credito")
    private Credito credito;

    @Column(name = "numero_cuota", nullable = false)
    private int numeroCuota;

    @Column(name = "valor_cuota", nullable = false, precision = 10, scale = 2)
    private BigDecimal valorCuota;

    @Column(name = "interes_pagado", nullable = false, precision = 10, scale = 2)
    private BigDecimal interesPagado;

    @Column(name = "capital_pagado", nullable = false, precision = 10, scale = 2)
    private BigDecimal capitalPagado;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal saldo;

    @Column(name = "fecha_vencimiento")
    private LocalDate fechaVencimiento;

    @Column(length = 15)
    private String estado; // PENDIENTE, PAGADA, VENCIDA, ANULADA

    // Getters y setters
    // ...

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Credito getCredito() {
        return credito;
    }

    public void setCredito(Credito credito) {
        this.credito = credito;
    }

    public int getNumeroCuota() {
        return numeroCuota;
    }

    public void setNumeroCuota(int numeroCuota) {
        this.numeroCuota = numeroCuota;
    }

    public BigDecimal getValorCuota() {
        return valorCuota;
    }

    public void setValorCuota(BigDecimal valorCuota) {
        this.valorCuota = valorCuota;
    }

    public BigDecimal getInteresPagado() {
        return interesPagado;
    }

    public void setInteresPagado(BigDecimal interesPagado) {
        this.interesPagado = interesPagado;
    }

    public BigDecimal getCapitalPagado() {
        return capitalPagado;
    }

    public void setCapitalPagado(BigDecimal capitalPagado) {
        this.capitalPagado = capitalPagado;
    }

    public BigDecimal getSaldo() {
        return saldo;
    }

    public void setSaldo(BigDecimal saldo) {
        this.saldo = saldo;
    }

    public LocalDate getFechaVencimiento() {
        return fechaVencimiento;
    }

    public void setFechaVencimiento(LocalDate fechaVencimiento) {
        this.fechaVencimiento = fechaVencimiento;
    }

    public String getEstado() {
        return estado;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }
}

File: PROYECTO/01.SERVIDOR/WS_DOTNET_SOAP_BanQuito/WS_DOTNET_SOAP_BanQuito/ec.edu.monster.model/CuotaAmortizacion.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WS_DOTNET_SOAP_BanQuito.ec.edu.monster.model
{
    [Table("cuota_amortizacion")]
    public class CuotaAmortizacion
    {
        [Key]
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("id_credito")]
        [ForeignKey("Credito")]
        public long IdCredito { get; set; }

        public virtual Credito Credito { get; set; }

        [Required]
        [Column("numero_cuota")]
        public int NumeroCuota { get; set; }

        [Required]
        [Column("valor_cuota", TypeName = "decimal")]
        public decimal ValorCuota { get; set; }

        [Required]
        [Column("interes_pagado", TypeName = "decimal")]
        public decimal InteresPagado { get; set; }

        [Required]
        [Column("capital_pagado", TypeName = "decimal")]
        public decimal CapitalPagado { get; set; }

        [Required]
        [Column("saldo", TypeName = "decimal")]
        public decimal Saldo { get; set; }

        [Column("fecha_vencimiento", TypeName = "date")]
        public DateTime? FechaVencimiento { get; set; }

        [Column("estado")]
        [StringLength(15)]
        public string Estado { get; set; } // PENDIENTE, PAGADA, etc.
    }
}
